### PR TITLE
test: add create react app project test

### PR DESCRIPTION
**Pull request by levithomason**
### Why?

We only test our components pre-distribution.  This test adds a test that consumes our package in a real project.  This allows us to catch errors that a user would see when building Stardust in their repo.

### How?

Local: `yarn test:projects:cra-ts`

CI: runs automatically

## What?

This PR adds a test that spins up a fresh React app using `create-react-app`.  It then runs a Stardust build, copies our `/dist` into the new React app and tries to compile a few components.

### Results

The current error we have in `0.5.1` is revealed:

```
node_modules/@types/react/index.d.ts(2432,13): error TS2717: Subsequent property
declarations must have the same type. Property 'time' must be of type
'DetailedHTMLProps<TimeHTMLAttributes<HTMLElement>, HTMLElement>', but here has type
'DetailedHTMLProps<TimeHTMLAttributes<HTMLElement>, HTMLElement>'.
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "pretest": "yarn satisfied && gulp build:docs:docgen",
     "test": "cross-env NODE_ENV=test jest --coverage",
     "test:visual": "gulp screener",
+    "test:projects:es": "rm -rf test/projects/ts-app && yarn link && cd test/projects && create-react-app ts-app --scripts-version=react-scripts-ts && cd ts-app && yarn link @stardust-ui/react && yarn build",
```
Comment (smykhailov) [inline package.json]: `rm` command doesn't exist on Windows machine

**Comment by smykhailov** [comment]
I'm afraid that there is the problem with the approach to link @stradust-ui/react in the ts-app. As in this case, we link it in the same folder structure as it is presented in the source code, not the one which we will distribute and we will not be able to catch the issues with relative paths imports in d.ts files

**Comment by hughreeling** [comment]
Fixed build on windows and updated dependencies to fix various errors related to bugs in @type/history etc.
Now runs and completes when run yarn test:projects:es.

I think we still need to add this as part of release - but please review changes first.

File: build/gulp/tasks/dist.ts [hunk 2]
```diff
@@ -18,13 +18,15 @@ task('clean:dist', cb => {
   rimraf(`${config.paths.dist()}/*`, cb)
 })
 
+task('clean:ts-app', cb => {
```
Comment (hughreeling) [inline build/gulp/tasks/dist.ts]: Is this right place for this?

**Comment by levithomason** [comment]
Seems CI doesn't like the `cd` command.  This works locally, so we'll have to get that sorted out:

```
================================================================================
CRA TS: Creating React App
================================================================================
/bin/sh: 1: cd: can't cd to /home/circleci/project/test/projects
[00:31:05] 'build:test:projects:cra-ts' errored after 10 ms
[00:31:05] Error: child process exited with code 2
```

**Comment by kuzhelov** [comment]
closing as the issue this PR targets was addressed by #248

**Pull request closed without merging**